bin/ghe-backup

```diff
@@ -159,59 +156,54 @@ bm_init > /dev/null
 ghe-backup-store-version  ||
 echo "Warning: storing backup-utils version remotely failed."
 
-echo "Backing up GitHub settings ..."
-ghe-backup-settings || failures="$failures settings"
-
-echo "Backing up SSH authorized keys ..."
-bm_start "ghe-export-authorized-keys"
-ghe-ssh "$GHE_HOSTNAME" -- 'ghe-export-authorized-keys' > authorized-keys.json ||
-failures="$failures authorized-keys"
-bm_end "ghe-export-authorized-keys"
-
-echo "Backing up SSH host keys ..."
-bm_start "ghe-export-ssh-host-keys"
-ghe-ssh "$GHE_HOSTNAME" -- 'ghe-export-ssh-host-keys' > ssh-host-keys.tar ||
-failures="$failures ssh-host-keys"
-bm_end "ghe-export-ssh-host-keys"
-
-echo "Backing up MySQL database ..."
-ghe-backup-mysql || failures="$failures mysql"
-
-echo "Backing up Redis database ..."
-ghe-backup-redis > redis.rdb || failures="$failures redis"
-
-echo "Backing up audit log ..."
-ghe-backup-audit-log || failures="$failures audit-log"
-
-echo "Backing up hookshot logs ..."
-ghe-backup-es-hookshot || failures="$failures hookshot"
-
-echo "Backing up Git repositories ..."
-ghe-backup-repositories || failures="$failures repositories"
-
-echo "Backing up GitHub Pages ..."
-ghe-backup-pages || failures="$failures pages"
-
-echo "Backing up storage data ..."
-ghe-backup-storage || failures="$failures storage"
+# Execute long running backup jobs in parallel
+parallel_commands="
+ghe-backup-settings
+ghe-ssh \"$GHE_HOSTNAME\" -- 'ghe-export-authorized-keys' > authorized-keys.json
+ghe-ssh \"$GHE_HOSTNAME\" -- 'ghe-export-ssh-host-keys' > ssh-host-keys.tar
+ghe-backup-mysql
```

Comment:
MySQL must always be backed up first and on its own to prevent consistency issues.

This ensures that the underlying database records exist for the repositories, storage objects, pages etc that are backed up later in the process.

Reply:
In addition to @snh 's point, with binary backup (xtrabackup) this will require us to have `ghe-backup-mysql` to run earlier than the rest.

Reply:
I am now backup up mysql first.